contracts/core/Dispatcher.sol

```diff
@@ -542,12 +556,39 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
     }
 
     // getOptimisticConsensusState
-    function getOptimisticConsensusState(uint256 height)
+    function getOptimisticConsensusState(uint256 height, uint256 clientId)
         external
         view
         returns (uint256 appHash, uint256 fraudProofEndTime, bool ended)
     {
-        return _lightClient.getState(height);
+        LightClient lightClient = _clientIdToLightClient[clientId];
+        if (address(lightClient) == address(0)) {
+            revert IBCErrors.lightClientNotFound(clientId);
+        }
+        return lightClient.getState(height);
+    }
+
+    // verify an EVM address matches an IBC portId.
+    // IBC_PortID = portPrefix + address (hex string without 0x prefix, case-insensitive)
+    // function portIdAddressMatch(address addr, string calldata portId) public view returns (bool isMatch) {
+    //     if (keccak256(abi.encodePacked(portPrefix)) != keccak256(abi.encodePacked(portId[0:portPrefixLen]))) {
+    //         return false;
+    //     }
+    //     string memory portSuffix = portId[portPrefixLen:];
+    //     isMatch = Ibc._hexStrToAddress(portSuffix) == addr;
+    // }
```

Comment:
remove?

Reply:
should be removed now in the commit named similarly to this one

Reply:
should be removed now in the commit named similarly to this one